### PR TITLE
docs: add UserWantToPlayList endpoint

### PR DESCRIPTION
**Pull request by ioslife**

### CODE REVIEW

File: docs/v1/get-user-want-to-play-list.md [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+
+:::
+
+## Source
+
+| Repo       | URL                                                                                                                  |
+| :--------- | :------------------------------------------------------------------------------------------------------------------- |
+| RAWeb      | https://github.com/RetroAchievements/RAWeb/blob/master/public/API/API_GetUserWantToPlayList.php                      |
+| api-js     | https://github.com/RetroAchievements/api-js/blob/main/src/user/getUserWantToPlayList.ts                              |
+| api-kotlin | https://github.com/RetroAchievements/api-kotlin/blob/main/src/main/kotlin/org/retroachivements/api/RetroInterface.kt |
```
Comment (ioslife) [inline docs/v1/get-user-want-to-play-list.md:124]: I wasn't sure about these two links

**Comment by ioslife** [comment]
RAWeb PR: https://github.com/RetroAchievements/RAWeb/pull/2549

File: .husky/pre-commit
Comment (wescopeland) [inline .husky/pre-commit:1]: Not sure what happened here, can we revert this?

File: docs/v1/get-user-want-to-play-list.md [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+} else if (response is NetworkResponse.Error) {
+    // if the server returns an error it be found here
+    val errorResponse: ErrorResponse? = response.body
+
+    // if the api (locally) had an internal error, it'll be found here
+    val internalError: Throwable? = response.error
+}
+```
+
+:::
```
Comment (wescopeland) [inline docs/v1/get-user-want-to-play-list.md:74]: Let's leave this as:

```md
## Client Library

Not yet supported.
```

For now.

File: docs/v1/get-user-want-to-play-list.md [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+      "title": "~Hack~ Knuckles the Echidna in Sonic the Hedgehog",
+      "imageIcon": "/Images/074560.png",
+      "consoleId": 1,
+      "pointsTotal": 1500,
+      "achievementsPublished": 50
+    }
+    // ...
+  ]
+}
+```
```
Comment (wescopeland) [inline docs/v1/get-user-want-to-play-list.md:114]: Safe to remove this

File: docs/v1/get-user-want-to-play-list.md (hunk 1 above)
Comment (wescopeland) [reply docs/v1/get-user-want-to-play-list.md:124]: Should be removed for the time being

File: .husky/pre-commit
Comment (ioslife) [reply .husky/pre-commit:1]: Finally got that fixed. I have no idea what is going on with that

**Comment by wescopeland** [comment]
Right now there's no way to access the page via the sidenav. It should also be linked in the README.md file.

**Review by wescopeland — APPROVED** [review]

File: docs/v1/get-user-want-to-play-list.md [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+{
+  "Count": 100,
+  "Total": 1287,
+  "Results": [
+    {
+      "GameID": 20246,
+      "Title": "~Hack~ Knuckles the Echidna in Sonic the Hedgehog",
+      "ImageIcon": "/Images/074560.png",
+      "ConsoleID": 1,
+      "PointsTotal": 1500,
```
Comment (wescopeland) [inline docs/v1/get-user-want-to-play-list.md:51]: ConsoleName should be documented here too.

https://github.com/RetroAchievements/RAWeb/pull/2549/files#diff-64244a633e1006f4f355365ef4af32e102627504f2f3056e5a056a22afba744fR59

File: docs/v1/get-user-want-to-play-list.md [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+{
+  "Count": 100,
+  "Total": 1287,
+  "Results": [
+    {
+      "GameID": 20246,
+      "Title": "~Hack~ Knuckles the Echidna in Sonic the Hedgehog",
+      "ImageIcon": "/Images/074560.png",
+      "ConsoleID": 1,
+      "ConsoleID": "Genesis/Mega Drive",
```
Comment (ioslife) [inline docs/v1/get-user-want-to-play-list.md:51]:
```suggestion
      "ConsoleName": "Genesis/Mega Drive",
```

**Review by wescopeland — APPROVED** [review]

File: docs/v1/get-user-want-to-play-list.md [hunk 6]
```diff
@@ -0,0 +1,66 @@
+<script setup>
+import SampleRequest from '../components/SampleRequest.vue';
+</script>
+
+# User Want to Play Lost
```
Comment (ioslife) [inline docs/v1/get-user-want-to-play-list.md:5]:
```suggestion
# User Want to Play List
```

File: docs/v1/get-user-want-to-play-list.md [hunk 7]
```diff
@@ -0,0 +1,66 @@
+<script setup>
+import SampleRequest from '../components/SampleRequest.vue';
+</script>
+
+# User Want to Play List
+
+A call to this endpoint will retrieve a giver user's want to play list, targeted by their username.
```
Comment (ioslife) [inline docs/v1/get-user-want-to-play-list.md:7]:
```suggestion
A call to this endpoint will retrieve a given user's want to play list, targeted by their username.
```

File: docs/v1/get-user-want-to-play-list.md [hunk 8]
```diff
@@ -0,0 +1,64 @@
+<script setup>
+import SampleRequest from '../components/SampleRequest.vue';
+</script>
+
+# User Want to Play List
+
+A call to this endpoint will retrieve a giver user's want to play list, targeted by their username. Results will only show if the user calling matches the username provided or both users follow each other on the website.
```
Comment (ioslife) [inline docs/v1/get-user-want-to-play-list.md:7]: this is a little wordy. Open to suggestions.

Comment (wescopeland) [reply docs/v1/get-user-want-to-play-list.md:7]: > A call to this endpoint will retrieve a given user's "Want to Play Games" list, targeted by their username. Results will only be returned if the target user is yourself, or if both you and the target user follow each other.

File: docs/v1/get-user-want-to-play-list.md [hunk 9]
```diff
@@ -0,0 +1,64 @@
...
+
+[[toc]]
+
+## On-site Representation
+
+A user's want to play list can be found on on the user's Want to Play Games page:
```
Comment (wescopeland) [inline docs/v1/get-user-want-to-play-list.md:13]:
```suggestion
The user's Want to Play Games list page looks like:
```

File: docs/v1/get-user-want-to-play-list.md [hunk 10]
```diff
@@ -0,0 +1,64 @@
...
+
+![Want to Play](/want-to-play-list.png)
```
Comment (wescopeland) [inline docs/v1/get-user-want-to-play-list.md:15]:
```suggestion
![Want to Play Games List](/want-to-play-list.png)
```

**Review by wescopeland — APPROVED** [review]

File: README.md [hunk 11]
```diff
@@ -43,6 +43,7 @@ Click the function names to open their complete docs on the docs site.
 - [Recently Played Games](https://api-docs.retroachievements.org/v1/get-user-recently-played-games.html) - Get a list of games a user has recently played.
 - [Summary](https://api-docs.retroachievements.org/v1/get-user-summary.html) - Get a user's profile metadata.
 - [Completed Games](https://api-docs.retroachievements.org/v1/get-user-completed-games.html) - Legacy endpoint. Get hardcore and softcore completion metadata about games a user has played.
+- [Want to Play List](https://api-docs.retroachievements.org/v1/get-user-want-to-play-list.html) - Get a user's list of Want to Play Games.
```
Comment (wescopeland) [inline README.md:46]:
```suggestion
- [Want to Play Games List](https://api-docs.retroachievements.org/v1/get-user-want-to-play-list.html) - Get a user's list of Want to Play Games.
```

File: docs/.vitepress/config.mts [hunk 12]
```diff
@@ -106,6 +106,10 @@ export default defineConfig({
             text: "Completed Games",
             link: "/v1/get-user-completed-games",
           },
+          {
+            text: "Want to Play List",
```
Comment (wescopeland) [inline docs/.vitepress/config.mts:110]:
```suggestion
            text: "Want to Play Games List",
```

**Review by wescopeland — APPROVED** [review]

**Pull request merged**